### PR TITLE
Cust 1693 user prefs not being cleared

### PR DESCRIPTION
**Pull request by jsmylnycky**
Fixed up prefs so they use app keys instead of ids across the board.
Since prefs (even bool) come back as strings, I put in a check to see if the string exists and its set to 'true' for proper validation.
Added in a new event for when intro pages are hidden, and wired the sidenav to subscribe to it so it can invalidate its pages and add the new page to its hidden list.

### CODE REVIEW

File: dist/peach.js [hunk 1]
```diff
@@ -460,7 +460,7 @@ return /******/ (function(modules) { // webpackBootstrap
 	      var deferred = $q.defer();
 	      is_core = _.isUndefined(is_core) ? $peachApp.isCoreApp() : is_core;
 	      var activeAccount = $peachSession.getActiveAccount();
-	      var activeApp = is_core ? 1 : $peachSession.getActiveApp();
+	      var activeApp = is_core ? 'core' : $peachSession.getActiveApp();
```
Comment (robraux) [inline dist/peach.js]: Why _'s? seems odd given everything is camelCase

File: dist/peach.js [hunk 2]
```diff
@@ -4403,6 +4399,18 @@ return /******/ (function(modules) { // webpackBootstrap
...
 	    }
 	  }, {
+	    key: 'activateHideIntroPageSub',
+	    value: function activateHideIntroPageSub() {
+	      var _this2 = this;
+
+	      this.$peach.event.subscribe(this.$peach.event.events.HIDE_INTRO_PAGE, function (appInfo) {
+	        if (_.has(appInfo, 'key')) {
+	          _this2.hiddenIntroPages[appInfo.key] = true;
+	          console.dir(_this2.hiddenIntroPages);
```
Comment (robraux) [inline dist/peach.js]: console.dir?

File: src/directives/sidenav/sidenav.controller.js [hunk 3]
```diff
@@ -157,6 +158,15 @@ class SidenavController {
     });
   }
 
+  activateHideIntroPageSub() {
+    this.$peach.event.subscribe(this.$peach.event.events.HIDE_INTRO_PAGE, (appInfo) => {
+      if(_.has(appInfo, 'key')) {
+        this.hiddenIntroPages[appInfo.key] = true;
+        console.dir(this.hiddenIntroPages);
```
Comment (robraux) [inline src/directives/sidenav/sidenav.controller.js]: console.dir?

**Comment by robraux** [comment]
:+1: except for console's. Mixed case seems odd as well.

File: dist/peach.js (hunk 1 above)
Comment (jsmylnycky) [inline dist/peach.js]: @robraux  Thanks...this was just one of those things we did a long time ago, and it wasn't ever updated.  I just replaced all is_core with isCore.

**Comment by jsmylnycky** [comment]
Removed the console.dir (cant believe I forgot that one) and fixed isCore camelcase.

**Comment by againsley** [comment]
:+1:

**Comment by darushka** [comment]
:+1:

**Comment by jsmylnycky** [comment]
Released v0.3.33

**Pull request closed without merging**